students/jstrauss123/lesson03/mailroom.py

```diff
@@ -1,72 +1,101 @@
+    print("")
+    # create donor_name_list
+    donor_name_list = [k for k,v in donorl]
+    # prompt for donor name
     full_name = input("Please provide full name of donor: ")
+    print("")
     while full_name == "list":
-        print(donor_list)
+        # print donor names
+        #print(donor_name_list)
+        for i,j in donorl:
+            print(i)
+        #prompt for donor name
+        print("")
         full_name = input("Please provide full name of donor: ")
-    if full_name in donor_list:
-        #print("full name entered is: ", full_name)
-        idx_num = donor_list.index(full_name)
-        #print("index is: ", idx_num)
-        #str1 = "donamt_" + str(idx_num)
-        #print("str1 value is: ", str1)
-        #print(" list value is equal to: ", str1[-1])
-        # if donor in list send thank you email
-        print("Dear {}, Thank you so much for your generous contribution".format(donor_list[idx_num]))
-    # if user not in donor_list add new donor and prompt for contribution amount
-    else: 
-        donor_list.append(full_name)
-        idx_num = donor_list.index(full_name)
-        new_donamt = input("Enter donation amount for new donor: ")
-        newamt = float(new_donamt)
-        print("Dear {}, Thank you so much for your generous contribution".format(donor_list[idx_num]))
-        # code here to create new contribution list for new donor
-        #donamt_<insert index here>.append(newamt)
-    #print("response is: ", response)    
-    #if response_ty == "list"
-    #    print(donor_list)
-    #if response_ty in donor_list:
+    if full_name not in donor_name_list:
+        #name does not exist, add to sequence
+        donorl.append((full_name, []))
+    # prompt for contribution amount and add to sequence on matching name
+    contrib_amt = float(input("Enter donation amount for donor: "))
+    for i,j in donorl:
+        if i == full_name:
+            j.append(contrib_amt)
+            print(donorl)
+    # send thank you email
+    print("Dear {}, Thank you so much for your generous contribution of ${}.".format(full_name, contrib_amt))
+    print("")
+    return donorl
 
 # function - create report
-def fun_report():
-    #arr6 = [['Joey', 30, 199.99], ['Jan', 36, 29999.97], ['Bob', 55, 999999.99]] 
-    donor_len = len(donor_list)
-    count1 = 0
+def create_report():
     print("")
+    sum = 0
+    donor_len = len(donor_list)
+    #print("donor_len = ", donor_len)
+    count = 0
+    while count < donor_len:
+        totamt = 0
+        contrib_amts = donor_list[count][1]
+        count += 1
+        donor_name = donor_list[count][0]
+        for amt in contrib_amts:
+            totamt = totamt + amt    
+        report_detail = report_detail +
+        # notes for tomorrow: try to append donor name, total amount, and number of donations to report_detail
+    #report_detail = (donor, totamt, num_of_donations)    
+    report_detail = (donor_name, totamt)    
+    print(report_detail)
```

Comment:
some hints:
* use `len` and `sum` functions instead of manually calculating the values
* use `sorted` to sort your list
* loop over sorted structure and print the report

Reply:
unable to get sum function working.
sum(contrib_amts)
TypeError: 'float' object is not callable.  

Searched online for suggestions, but unable to resolve. Stuck with existing iteration for now. New code uploaded (new pull request).
https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/1249